agent/src/api/v1/proxy.rs

```diff
@@ -22,35 +23,40 @@ pub async fn relay_request(
     service_name: &str,
     source_instance_id: &Uuid,
 ) -> Result<Response, ServalError> {
-    let node_info = discover_service(service_name).await.map_err(|err| {
-        log::warn!("proxy_unavailable_services failed to find a node offering the service; service={service_name}; err={err:?}");
+    let mesh = MESH.get().expect("Peer network not initialized!");
+    let Some(peer) = mesh.find_role(&service_name.to_string()).await else {
+        log::warn!("proxy_unavailable_services failed to find a node offering the service; service={service_name}");
         metrics::increment_counter!("proxy:no_service");
-        err
-    })?;
+        return Err(ServalError::ServiceNotFound);
+    };
 
-    let result = proxy_request_to_other_node(req, &node_info, source_instance_id).await;
+    let result = proxy_request_to_other_node(req, &peer, source_instance_id).await;
     result.map_err(|err| {
-        log::warn!("Failed to proxy request to other node; node={node_info:?}; err={err:?}");
+        log::warn!("Failed to proxy request to peer; peer={peer:?}; err={err:?}");
         metrics::increment_counter!("proxy:failure");
         err
     })
 }
 
 async fn proxy_request_to_other_node(
     req: &mut Request<Body>,
-    info: &ServiceInfo,
+    peer: &PeerMetadata,
     source_instance_id: &Uuid,
 ) -> Result<Response, ServalError> {
-    let target_instance_id = get_service_instance_id(info)?;
-    let host = info.get_addresses().iter().next().unwrap(); // unwrap is safe because discover_service will never return a service without addresses
-    let port = info.get_port();
+    let target_instance_id = peer.name();
+    let Some(socket_addr) = peer.address() else {
```

Comment:
I am making this comment in the wrong place, but why does `peer.address() return an `Option`?